src/hcf-release/src/acceptance-tests-brain/test-scripts/008_cf_usb_test.sh

```diff
@@ -0,0 +1,199 @@
+    cf unbind-running-security-group internal-services-workaround
+    cf unbind-staging-security-group internal-services-workaround
+    cf delete-shared-domain -f ${CF_TCP_DOMAIN}
+
+    rm -rf "${TMP}"
+
+    set -o errexit
+    login_cleanup
+}
+trap test_cleanup EXIT ERR
+
+# --(0)-- Initialize tcp routing
+
+cf delete-shared-domain -f ${CF_TCP_DOMAIN} || true
+cf create-shared-domain    ${CF_TCP_DOMAIN} --router-group default-tcp
+cf update-quota default --reserved-route-ports -1
+
+# --(0.1) -- Initialize a security group to allow for inter-app comms
+# Attention: This SG opens the entire internal kube service network.
+
+#echo > ${TMP}/internal-services.json '[{ "destination": "192.168.77.77/32", "protocol": "all" }]'
+echo > ${TMP}/internal-services.json '[{ "destination": "0.0.0.0/0", "protocol": "all" }]'
+
+cf create-security-group       internal-services-workaround ${TMP}/internal-services.json
+cf bind-running-security-group internal-services-workaround
+cf bind-staging-security-group internal-services-workaround
+
+## --(1)-- Create and configure the mysql server
+
+cf push --no-start --no-route --health-check-type none ${SERVER_APP} -o mysql/mysql-server
+cf map-route ${SERVER_APP} ${CF_TCP_DOMAIN} --random-port | tee ${TMP}/mysql
+cf set-env   ${SERVER_APP} MYSQL_ROOT_PASSWORD ${MYSQL_PASS}
+cf set-env   ${SERVER_APP} MYSQL_ROOT_HOST '%'
+cf start     ${SERVER_APP}
+
+MYSQL_PORT=$(get_port mysql)
+
+wait_on_port ${MYSQL_PORT}
+
+## --(2)-- Create and configure the mysql client sidecar for usb.
+
+cf push ${SIDECAR_APP} --no-start -o splatform/cf-usb-sidecar-dev-mysql
+
+# Use a secret key that will be used by the USB to talk to your
+# sidecar, and set the connection parameters for the mysql client
+# sidecar so that it can talk to the mysql server from the previous
+# step.
+cf set-env ${SIDECAR_APP} SIDECAR_API_KEY ${SIDECAR_API_KEY}
+cf set-env ${SIDECAR_APP} SERVICE_MYSQL_HOST ${CF_TCP_DOMAIN}
+cf set-env ${SIDECAR_APP} SERVICE_MYSQL_PORT ${MYSQL_PORT}
+cf set-env ${SIDECAR_APP} SERVICE_MYSQL_USER ${MYSQL_USER}
+cf set-env ${SIDECAR_APP} SERVICE_MYSQL_PASS ${MYSQL_PASS}
+cf start   ${SIDECAR_APP}
+
+# --(3)-- Create a driver endpoint to the mysql sidecar (== service type)
+# Note that the -c ":" is required as a workaround to a known issue
+cf usb create-driver-endpoint ${SERVICE_TYPE} \
+    https://${SIDECAR_APP}.${CF_DOMAIN} ${SIDECAR_API_KEY} \
+    -c ":"
+
+# --(4)-- Check that the service is available in the marketplace and use it
+
+## Note: The commands without grep filtering are useful in case of
+## failures, providing immediate information about the data which runs
+## through and fails the filter.
+
+cf marketplace
+cf marketplace | grep ${SERVICE_TYPE}
+
+cf create-service ${SERVICE_TYPE} default ${SERVICE_INSTANCE}
+
+cf services
+cf services | grep ${SERVICE_INSTANCE}
+
+# -- If we want to, we can now create and push an app which uses the
+#    service-instance as database, and verify that it works.
```

Comment:
Umm, yes, we should probably do that. Otherwise we're only testing that the service registers, not that it works.

Reply:
Well, the test case is validating that the USB works, here, and creating a test service instance is enough for that. It goes through the USB to the sidecar and the sidecar talked properly to the database.

I don't think that having the service (mysql) work is part of the test scope here.

Reply:
Eh, alright.